packages/@aws-cdk/aws-elasticsearch/README.md

```diff
@@ -1,12 +1,160 @@
+This method accepts a domain endpoint of an already existing domain:
+
+```ts
+const domainEndpoint = 'https://my-domain-jcjotrt6f7otem4sqcwbch3c4u.us-east-1.es.amazonaws.com';
+const domain = Domain.fromDomainEndpoint(this, 'ImportedDomain', domainEndpoint);
+```
+
+### Permissions
+
+#### IAM
+
+Helper methods also exist for managing access to the domain.
+
+```ts
+const lambda = new lambda.Function(this, 'Lambda', { /* ... */ });
+
+// Grant write access to the app-search index
+domain.grantIndexWrite('app-search', lambda);
+
+// Grant read access to the 'app-search/_search' path
+domain.grantPathRead('app-search/_search', lambda);
+```
+
+### Encryption
+
+The domain can also be created with encryption enabled:
+
+```ts
+const domain = new es.Domain(this, 'Domain', {
+    version: es.ElasticsearchVersion.V7_4,
+    ebs: {
+        volumeSize: 100,
+        volumeType: EbsDeviceVolumeType.GENERAL_PURPOSE_SSD,
+    },
+    nodeToNodeEncryption: true,
+    encryptionAtRest: {
+        enabled: true,
+    },
+});
+```
+
+This sets up the domain with node to node encryption and encryption at
+rest. You can also choose to supply your own KMS key to use for encryption at
+rest.
+
+### Metrics
+
+Helper methods exist to access common domain metrics for example:
+
+```ts
+const freeStorageSpace = domain.metricFreeStorageSpace();
+const masterSysMemoryUtilization = domain.metric('MasterSysMemoryUtilization');
+```
+
 This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aws-cdk) project.
+
+### Fine grained access control
+
+The domain can also be created with a master user configured. The password can
+be supplied or dynamically created if not supplied.
+
+```ts
+const domain = new es.Domain(this, 'Domain', {
+    version: es.ElasticsearchVersion.V7_1,
+    enforceHttps: true,
+    nodeToNodeEncryption: true,
+    encryptionAtRest: {
+        enabled: true,
+    },
+    fineGrainedAccessControl: {
+        masterUserName: 'master-user',
+    },
+});
+
+const masterUserPassword = domain.masterUserPassword;
+```
+
+### Using unsigned basic auth
+
+For convenience, the domain can be configured to allow unsigned HTTP requests
+that use basic auth. Unless the domain is configured to be part of a VPC this
+means anyone can access the domain using the configured master username and
+password.
+
+To enable unsigned basic auth access the domain is configured with an access
+policy that allows anyonmous requests, HTTPS required, node to node encryption,
+encryption at rest and fine grained access control.
+
+If the above settings are not set they will be configured as part of enabling
+unsigned basic auth.
+
+If no master user is configured a default master user is created with the
+username `admin`.
+
+```ts
+const domain = new es.Domain(this, 'Domain', {
+    version: es.ElasticsearchVersion.V7_1,
+    useUnsignedBasicAuth: true,
+});
+
+const masterUserPassword = domain.masterUserPassword;
```

Comment:
Add that the password is stored as a secret in secrets manager and mention the name of the secret. Users would have to look it up in order to connect to the cluster manually.

Reply:
Done.